actor/builtin/miner/miner.go

```diff
@@ -461,6 +465,36 @@ func (ma *Actor) IsBootstrapMiner(ctx exec.VMContext) (bool, uint8, error) {
 	return ma.Bootstrap, 0, nil
 }
 
+const (
+	WithinProvingPeriod = iota
+	AfterProvingPeriod
+	AfterGenerationAttackThreshold
```

Comment:
This enumeration is a good idea. A few suggestions:

1. Please move this definition up with the other constants.
2. This struct already defines several constants. Please prefix with 'PoStState', e.g. `PoStStateWithinProvingPeriod`.
3. There should be a separate state for the state when the miner is not proving any storage. `PoStStateNoStorage`?